### DISCUSSION TITLE
Unable to get Docker Installation Working

### DISCUSSION THREAD

**Discussion by RickoT** [postalserver/postal#1844] [Help with using Postal]
Greetings!

I am unable to get the docker installation to work.  

Even just running the postal bootstrap command simply returns the following 
![image](https://user-images.githubusercontent.com/2433564/152629720-952352ba-ecc8-4dc2-bc3f-76dac33acde0.png)

I was able to grab the sample config file and make the changes i want, but now when I run the docker container, it just gets marked "completed" and starts a new one

Any assistance would be appreciated.

Thanks!
~Rick

**Comment by willpower232** [comment]
It sounds like you might not be following the official install instructions because you shouldn't have Postals code just on your server any more.

Have a look at https://docs.postalserver.io/

**Comment by RickoT** [comment]
Hi WillPower,

According to the prereqs, you need to clone the repository to bootstrap everything.  Is this not the case anymore?  Perhaps the documentation needs to be updated?  Is there an official docker documentation piece that I am missing?

https://docs.postalserver.io/install/prerequisites

Thanks for your help!
~Rick

**Reply by willpower232** [reply]
I can tell from your screenshot that you've cloned [postalserver/postal](https://github.com/postalserver/postal) instead of [postalserver/install](https://github.com/postalserver/install) or whatever the instructions say

**Reply by RickoT** [reply]
> I can tell from your screenshot that you've cloned [postalserver/postal](https://github.com/postalserver/postal) instead of [postalserver/install](https://github.com/postalserver/install) or whatever the instructions say

Ahhhhh!!! Thank you for pointing out my mistake.  I will check that out tomorrow and mark that as an answer if im good

Thanks!

**Reply by RickoT** [reply]
Hey WillPower, so that was the trick for that.  However, now that I have it all up and running, when I try to create a new server I get an error 500.  Where are some logs I can provide?

Also, do I have to use the /opt/postal/bin/postal binary to stop and start all this or can I use something like portainer and monitor it there?  I would prefer to not have to leverage a second binary.

Thanks!
~Rick

**Reply by willpower232** [reply]
re the logs, you need to get them from the container, you'll be interested in #1845 

The new postal "binary" is just a thin wrapper around docker-compose, as long as you're running (at least) one of each process, you can use whatever you want. Sticking to docker-compose makes it easier to get started with but if you know how to lay it out in portainer then go for it.

**Reply by RickoT** [reply]
Ah thanks for the reference.  I put my answer in that post as I believe I have resolved this for my purposes.  Thank you for everything else, you saved my bacon and helped me figure this out

https://github.com/postalserver/postal/discussions/1845#discussioncomment-2142263

Link for posterity.

**Comment by RickoT** [comment]
I did have one last question.  So the runner container does not seem to work,  Portainer will not let me use the "profiles" property, and when I run it without it just seems to run the postal command and shutdown.  Is this container necessary?   Can you think of a reason why the profiles property will not work? (If necessary I can create a new discussion)

Thanks! 
~Rick

**Reply by willpower232** [reply]
when you say runner container, what do you mean?

**Reply by RickoT** [reply]
This container is in the created docker-compose.yml file

  runner:
    profiles: ["tools"]
    image: ghcr.io/postalserver/postal:{{version}}
    command: postal
    network_mode: host
    volumes:
      - /opt/postal/config:/config

**Reply by willpower232** [reply]
does {{version}} get replaced with a real tag? https://github.com/postalserver/postal/pkgs/container/postal

**Reply by RickoT** [reply]
I replaced version with 2.1.0, but when the container runs it just shows the postal command options, then the container dies because it has nothing to do

**Reply by willpower232** [reply]
You'll notice in the template docker-compose file, each command is different as each process is run separately

https://github.com/postalserver/install/blob/main/templates/docker-compose.yml

In v2, the ability to run all the processes from a single command was removed.

**Reply by RickoT** [reply]
Yeah, that's the source docker-compose.yml file I used to set up my stack in portainer, but the last one (runner) just runs the postal command, then shuts down because the postal command just displays the arguments then exits.

**Reply by willpower232** [reply]
Ah I hadn't noticed that before. Looks like it is for initialising and upgrading so doesn't need to be kept running. Postal only needs the first five for actual usage.
